src/cdk/v2/destinations/the_trade_desk/transform.js

```diff
@@ -0,0 +1,39 @@
+const lodash = require('lodash');
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const { processRecordInputs } = require('./transformRecord');
+const { processConversionInputs } = require('./transformConversion');
+const { handleRtTfSingleEventError } = require('../../../../v0/util');
+const { EventType } = require('../../../../constants');
+
+const processRouterDest = async (inputs) => {
+  const respList = [];
+  const { destination } = inputs[0];
+  const groupedInputs = lodash.groupBy(inputs, (input) => input.message.type);
+
+  Object.keys(groupedInputs).forEach(async (type) => {
```

Comment:
I don't think forEach supports async await properly

Reply:
used map for async/await